server/sonar-web/src/main/js/components/controls/Dropdown.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+
+interface RenderProps {
+  closeDropdown: () => void;
+  onToggleClick: (event: React.SyntheticEvent<HTMLElement>) => void;
+  open: boolean;
+}
+
+interface Props {
+  children: (renderProps: RenderProps) => JSX.Element;
+  onOpen?: () => void;
+}
+
+interface State {
+  open: boolean;
+}
+
+export default class Dropdown extends React.PureComponent<Props, State> {
+  toggleNode?: HTMLElement;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = { open: false };
+  }
+
+  componentDidUpdate(_: Props, prevState: State) {
+    if (!prevState.open && this.state.open) {
+      this.addClickHandler();
+      if (this.props.onOpen) {
+        this.props.onOpen();
+      }
+    }
+
+    if (prevState.open && !this.state.open) {
+      this.removeClickHandler();
+    }
+  }
+
+  componentWillUnmount() {
+    this.removeClickHandler();
+  }
+
+  addClickHandler = () => {
+    window.addEventListener('click', this.handleClick);
+  };
+
+  removeClickHandler = () => {
+    window.removeEventListener('click', this.handleClick);
+  };
+
+  handleClick = (event: MouseEvent) => {
```

Comment:
I would rename this to something a bit more explicit like `handleWindowClick` to make it more obvious that it's not just to handle any click on the element.